src/juju/index.js

```diff
@@ -15,18 +17,28 @@ const options = {
 
 const controllerURL = process.env.REACT_APP_CONTROLLER_URL;
 
+/**
+  Connects to the controller at the url defined in the REACT_APP_CONTROLLER_URL
+  environment variable.
+  @returns {Object} conn The controller connection instance.
+*/
 async function loginWithBakery() {
-  try {
-    const juju = await jujulib.connect(controllerURL, options);
-    const conn = await juju.login({});
+  const juju = await jujulib.connect(controllerURL, options);
+  const conn = await juju.login({});
+  return conn;
+}
+
+/**
+  Fetches the model list from the supplied Juju controller.
+  @param {Object} conn The controller connection instance.
+  @returns {Object} models The list of model objects under the key `userModels`.
+ */
+function fetchModelList(conn) {
+  return async function(dispatch) {
     const modelManager = conn.facades.modelManager;
     const models = await modelManager.listModels({ tag: conn.info.identity });
-    console.log("models", models);
-  } catch (error) {
-    // XXX Surface error to UI.
-    console.log("unable to connect:", error);
-    return;
-  }
+    dispatch(actions.updateModelList(models.userModels));
```

Comment:
I'm wondering if maybe the reducer should get the raw data from the controller? 

ie) `dispatch(actions.updateModelList(models));`

Reply:
Not sure what you mean here, maybe let's chat about it on irc.